test/e2e/choice_test.go

```diff
@@ -0,0 +1,147 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	eventingduckv1alpha1 "github.com/knative/eventing/pkg/apis/duck/v1alpha1"
+	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	v1alpha1 "github.com/knative/eventing/pkg/apis/messaging/v1alpha1"
+	eventingtesting "github.com/knative/eventing/pkg/reconciler/testing"
+	"github.com/knative/eventing/test/base/resources"
+	"github.com/knative/eventing/test/common"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	pkgTest "knative.dev/pkg/test"
+)
+
+type caseConfig struct {
+	filter bool
+}
+
+func TestChoice(t *testing.T) {
+	const (
+		senderPodName = "e2e-choice"
+	)
+	table := []struct {
+		name        string
+		casesConfig []caseConfig
+		expected    string
+	}{
+		{
+			name: "two-cases-pass-first-case-only",
```

Comment:
Why not make these two separate test cases? That way we can remove a for loop below:

```golang
	table := []struct {
		name        string
		casesConfig caseConfig
		expected    string
	}{
		{
			name: "don't filter",
			casesConfig: caseConfig{
				filter: false,
			},
		},
		{
			name: "filter",
			casesConfig: caseConfig{
				filter: true,
			},
			expected: "choice-filter",
		},
	}
```

Reply:
One choice can have multiple cases. It's a bit confusing because there is only one test (more to come).